src/main/app/src/app/informatie-objecten/informatie-object-edit/informatie-object-edit.component.ts

```diff
@@ -89,15 +89,23 @@ export class InformatieObjectEditComponent implements OnInit {
                                                  .validators(Validators.required)
                                                  .build();
 
-        const status = new SelectFormFieldBuilder().id('status').label('status')
-                                                   .validators(Validators.required)
-                                                   .optionLabel('label').options(informatieobjectStatussen)
-                                                   .value({
-                                                       label: this.translateService.instant(
-                                                           'informatieobject.status.' + this.infoObject.status.toUpperCase()),
-                                                       value: this.infoObject.status.toUpperCase()
-                                                   })
-                                                   .build();
+        let status;
+        if (this.infoObject.status) {
+            status = new SelectFormFieldBuilder().id('status').label('status')
+                                                       .validators(Validators.required)
+                                                       .optionLabel('label').options(informatieobjectStatussen)
+                                                       .value({
+                                                           label: this.translateService.instant(
+                                                               'informatieobject.status.' + this.infoObject.status.toUpperCase()),
+                                                           value: this.infoObject.status.toUpperCase()
+                                                       })
+                                                       .build();
+        } else {
```

Comment:
Er is onnodig code gedupliceerd. Het .value gedeelte is het enige wat afwijkt. Dus dat is het enige wat in een if hoeft te staan,

Reply:
als je dit refactored dan geeft dit een fout in intellij dat je dingen mist op build, daarom heb ik de changes ge-approved
maar kijk maar even marco.

Reply:
Had al een inline if constructie geprobeerd met undefined en {} mee te geven als this.infoObject.status niet gevuld was. Maar dat gaf het selectfield toch de impressie dat hij gevuld was, waardoor je het document kon aanpassen zonder de verplichte status in te vullen. Zonet nog even geprobeerd als ik hier null aan meegeef, dat werkte wel. Dus heb deze if constructie eruit kunnen halen.